lib/components/Charts/LineChart/index.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  ChartConfig,
+  ChartContainer,
+  ChartTooltip,
+  ChartTooltipContent,
+} from "@/ui/chart"
+import { ForwardedRef, forwardRef } from "react"
+import {
+  CartesianGrid,
+  Line,
+  LineChart as LineChartPrimitive,
+  XAxis,
+} from "recharts"
+import { autoColor } from "../utils/colors"
+
+type ChartItem<LineKeys extends string> = {
+  label: string
+  values: Record<LineKeys, number>
+}
+
+type AxisConfig = {
+  hide?: boolean
+  tickFormatter?: (value: string) => string
+}
+
+export type LineChartConfig<Keys extends string = string> = Record<
+  Keys,
+  ChartConfig[string]
+>
```

Comment:
We can probably reuse this type across charts!

Reply:
Creating a common type, right? Let me try it!